src/lib.rs

```diff
@@ -88,7 +88,13 @@ impl<const ORDER: usize> Heap<ORDER> {
 
         while current_start + size_of::<usize>() <= end {
             let lowbit = current_start & (!current_start + 1);
-            let size = min(lowbit, prev_power_of_two(end - current_start));
+            let mut size = min(lowbit, prev_power_of_two(end - current_start));
+            
+            // If the order of size is larger than the max order,
+            // split it into smaller blocks.
+            if size.trailing_zeros() as usize > (ORDER - 1) {
```

Comment:
I think comparing size and ORDER is easier and faster here?

Reply:
Since `trailing_zeros` calculation is calculated again when adding a block to `free_list`, I think we can calculate it first for comparison, if the block needs to be split, just simply assign it as the maximum ORDER, I think this should be faster. (I've forced-push this change)